src/devtools/store.js

```diff
@@ -769,7 +777,16 @@ export default class Store extends EventEmitter {
           }
 
           element = ((this._idToElement.get(id): any): Element);
+          const prevChildren = element.children;
           element.children = Array.from(children);
+          if (element.children.length !== prevChildren.length) {
+            throw new Error(
+              'Fiber ' +
+                id +
+                ' received a different number of children on reorder. ' +
+                'This is a bug in React DevTools.'
+            );
+          }
```

Comment:
I was  adding a version of this locally too that also compared contents to make sure nothing new was added, but I hadn't pushed it yet. I'll just rebase on top of yours! 😁

Reply:
24736e4ff57832b9fcc1da8137729ae65a78124a

https://github.com/bvaughn/react-devtools-experimental/blob/60b614f5272fd0fa400c555e011f8262582c5b55/src/devtools/store.js#L776-L787